script/testing/junit/src/FilterTrace.java

```diff
@@ -0,0 +1,154 @@
+//                    skip_flag = true;
+//                    writeToFile(writer, constants.SKIP);
+//                    break;
+//                }
+//            }
+            // the code below remove the queries that contain any skip keyword
+            for(String skip_word:skip_list) {
+                if (cur_sql.contains(skip_word)) {
+                    mog.comments.clear();
+                    skip_flag = true;
+                    break;
+                }
+            }
+            if(skip_flag){
+                skip_flag = false;
+                continue;
+            }
+            // filter out nested SELECT statements
+            if(getFrequency(cur_sql, "SELECT")>1){
+                mog.comments.clear();
+                continue;
+            }
+            writeToFile(writer, mog.queryFirstLine);
+            writeToFile(writer, cur_sql);
+            if(skip_flag){
+                if(mog.queryFirstLine.contains(constants.QUERY)){
+                    writeToFile(writer, constants.SEPARATION);
+                    if(mog.queryResults.size()>0){
+                        if(mog.queryResults.get(0).contains(constants.VALUES)){
+                            writeToFile(writer, mog.queryResults.get(0));
+                        }else{
+                            for(String i:mog.queryResults){
+                                writeToFile(writer, i);
+                            }
+                        }
+
+                    }
+                }
+            }else{
+                if(mog.queryFirstLine.contains(constants.QUERY)){
+                    writeToFile(writer, constants.SEPARATION);
+                    try{
+                        Statement statement = conn.createStatement();
+                        statement.execute(cur_sql);
+                        ResultSet rs = statement.getResultSet();
+                        List<String> res = mog.processResults(rs);
+                        if(res.size()==0){
+                            writer.write('\n');
+                        }else{
+                            String hash = TestUtility.getHashFromDb(res);
+                            String queryResult = res.size() + " values hashing to " + hash;
+                            writeToFile(writer, queryResult);
+                        }
+                    }catch(Throwable e){
+                        throw new Throwable(e.getMessage() + ": " + cur_sql);
+                    }
+                }else{
+                    try{
+                        Statement statement = conn.createStatement();
+                        statement.execute(cur_sql);
+                    }catch(Throwable e){
+                        throw new Throwable(e.getMessage() + ": " + cur_sql);
+                    }
+                }
+            }
+            writer.write('\n');
+            skip_flag = false;
+            mog.comments.clear();
+            mog.queryResults.clear();
+        }
+        writer.close();
```

Comment:
I'm not sure if you're familiar with try-with-resources (https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html), but it tends to be a safer/cleaner way to manage closable resources. It may not be needed for a short lived script like this, but it's handy to know.